platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java

```diff
@@ -0,0 +1,305 @@
+     * Creates an {@link org.apache.solr.client.solrj.SolrClient} with the default http address
+     * url.
+     *
+     * @return SolrClient
+     */
+    static SolrClient getHttpSolrClient() {
+        return new HttpSolrClient(getDefaultHttpAddress());
+    }
+
+    @Override
+    public Future<SolrClient> newClient(String core) {
+        ConfigurationStore configStore = ConfigurationStore.getInstance();
+        if (System.getProperty("solr.data.dir") != null) {
+            configStore.setDataDirectoryPath(System.getProperty("solr.data.dir"));
+        }
+        String solrUrl = System.getProperty("solr.http.url", getDefaultHttpsAddress());
+        return getHttpSolrClient(solrUrl, core);
+    }
+
+    public static Future<SolrClient> getHttpSolrClient(String url) {
+        return getHttpSolrClient(url, DEFAULT_CORE_NAME, null);
+    }
+
+    public static Future<SolrClient> getHttpSolrClient(String url, String coreName) {
+        return getHttpSolrClient(url, coreName, null);
+    }
+
+    public static Future<SolrClient> getHttpSolrClient(String url, String coreName,
+            String configFile) {
+        String solrUrl = StringUtils.defaultIfBlank(url, SystemBaseUrl.constructUrl("/solr"));
+        String coreUrl = url + "/" + coreName;
+
+        RetryPolicy retryPolicy = new RetryPolicy()
+                .withBackoff(10, TimeUnit.MINUTES.toMillis(1), TimeUnit.MILLISECONDS);
+        return Failsafe.with(retryPolicy)
+                .with(executor)
+                .onRetry((c, failure, ctx) -> LOGGER.debug(
+                        "Attempt {} failed to create HTTP Solr client ({}). Retrying again.",
+                        ctx.getExecutions(),
+                        coreName))
+                .onFailedAttempt(failure -> LOGGER.debug(
+                        "Attempt failed to create HTTP Solr client (" + coreName + ")",
+                        failure))
+                .onSuccess(client -> LOGGER.debug("Successfully created HTTP Solr client ({})",
+                        coreName))
+                .onFailure(failure -> LOGGER.warn(
+                        "All attempts failed to create HTTP Solr client (" + coreName + ")",
+                        failure))
+                .get(() -> createSolrHttpClient(solrUrl, coreName, configFile, coreUrl));
+    }
+
+    private static SolrClient createSolrHttpClient(String url, String coreName, String configFile,
+            String coreUrl) throws IOException, SolrServerException {
+        SolrClient client;
+        if (StringUtils.startsWith(url, "https")) {
+            createSolrCore(url, coreName, configFile, getHttpClient(false));
+            client = new HttpSolrClient(coreUrl, getHttpClient(true));
+        } else {
+            createSolrCore(url, coreName, configFile, null);
+            client = new HttpSolrClient(coreUrl);
+        }
+        return client;
+    }
+
+    private static CloseableHttpClient getHttpClient(boolean retryRequestsOnError) {
```

Comment:
Should we name this method `getHttpsClient` to make it clear that it will create an SSL http client?